src/framework/components/anim/component.js

```diff
@@ -228,7 +237,8 @@ class AnimComponent extends Component {
             transitions,
             this._parameters,
             this._activate,
-            this
+            this,
+            this._resetTrigger.bind(this)

```

Comment:
if you implement consumedTriggers as a set, I'd simply pass the set to controller here to add to it as needed, instead of doing this callback.